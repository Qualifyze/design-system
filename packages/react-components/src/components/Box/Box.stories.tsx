import React from 'react';

import { Box } from './Box';

export default {
  title: 'Components/Box',
  component: Box,
};

export const basic = () => (
  <Box css={{ padding: '$2', bc: '$indigo4' }}>Hello</Box>
);
