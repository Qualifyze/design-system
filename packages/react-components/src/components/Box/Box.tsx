/** @jsxImportSource theme-ui */
import * as React from 'react';
import { Box as UIBox, BoxProps as UIBoxProps } from 'theme-ui';

export interface BoxProps extends UIBoxProps {}

export const Box = React.forwardRef<HTMLDivElement, BoxProps>((props, ref) => {
  return <UIBox ref={ref} {...props} />;
});
