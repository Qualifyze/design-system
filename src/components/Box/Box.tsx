export { Box } from '@theme-ui/components';
