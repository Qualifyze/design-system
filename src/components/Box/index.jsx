/* eslint-disable no-shadow */
/* eslint-disable prefer-template */
/* eslint-disable no-underscore-dangle */
import styled from '@emotion/styled'
import css, { get } from '@styled-system/css'
import shouldForwardProp from '@styled-system/should-forward-prop'

import {
  compose,
  space,
  layout,
  typography,
  color,
  position,
  flexbox,
  grid,
  sx,
} from '../../util/style'

const base = props => css(props.__css)(props.theme)
const variant = ({ theme, variant, tx = 'variants' }) =>
  css(get(theme, tx + '.' + variant, get(theme, variant)))(theme)

const Box = styled('div', {
  shouldForwardProp,
})(
  {
    boxSizing: 'border-box',
    position: 'relative',
    margin: 0,
    minWidth: 0,
  },
  base,
  variant,
  sx,
  props => props.css,
  compose(space, layout, typography, color, position, flexbox, grid)
)

export default Box
