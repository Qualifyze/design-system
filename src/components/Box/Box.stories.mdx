import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { Box } from './Box';

<Meta title="Components/Box" component={Box} />

# Box

The heart of our design system.

<Canvas>
  <Story name="Box">
    <Box sx={{ p: 4, bg: 'text' }}>hello</Box>
  </Story>
</Canvas>