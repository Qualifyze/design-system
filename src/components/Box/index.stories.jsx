import React from 'react'
import { text } from '@storybook/addon-knobs'

import Box from './index'

export default { title: 'Box', component: Box }

export const Default = () => {
  const bg = text('Background', 'primary.200')
  const m = text('Margin', '2')
  const p = text('Padding', '4')

  return (
    <Box bg={bg} m={m} p={p}>
      Hola, mundo!
    </Box>
  )
}
Default.story = {
  name: 'default',
}
