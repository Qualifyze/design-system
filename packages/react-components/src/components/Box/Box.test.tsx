import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Box } from './Box';

describe('Box', () => {
  test('renders', () => {
    render(<Box>hola</Box>);
    expect(screen.getByText(/hola/)).toBeInTheDocument();
  });

  test('renders with css prop', () => {
    render(<Box css={{ paddingTop: 1 }}>hola</Box>);
    const B = screen.getByText(/hola/);
    // The `toHaveStyle` matcher relies on a serializer of the created CSS,
    // which we don't have for Stitches yet
    // expect(B).toHaveStyle(`padding-top`, `1px`);
    expect(B).toMatchSnapshot();
  });

  test('renders with theme-aware values from Stitches config', () => {
    render(<Box css={{ color: `$gray12` }}>hola</Box>);
    const B = screen.getByText(/hola/);
    // The `toHaveStyle` matcher relies on a serializer of the created CSS,
    // which we don't have for Stitches yet
    // expect(B).toHaveStyleRule(`color`, `var(--theme-ui-colors-positive)`);
    // expect(B).toHaveStyleRule(
    //   `background-color`,
    //   `var(--theme-ui-colors-caution)`
    // );
    expect(B).toMatchSnapshot();
  });

  test('passes `data-` attributes through to the DOM', () => {
    render(<Box data-testid="some-random-id">hola</Box>);
    const B = screen.getByText(/hola/);
    expect(B).toHaveAttribute(`data-testid`, `some-random-id`);
    expect(B).toMatchSnapshot();
  });

  test('passes `aria-` attributes through to the DOM', () => {
    render(<Box aria-hidden="true">hola</Box>);
    const B = screen.getByText(/hola/);
    expect(B).toHaveAttribute(`aria-hidden`, `true`);
    expect(B).toMatchSnapshot();
  });

  test('renders as passed HTML element', () => {
    render(<Box as="h1">hola</Box>);
    const B = screen.getByRole(`heading`);
    expect(B).toMatchSnapshot();
  });
});
