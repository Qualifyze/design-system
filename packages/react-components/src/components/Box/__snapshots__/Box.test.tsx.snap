// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box passes \`aria-\` attributes through to the DOM 1`] = `
<div
  aria-hidden="true"
  class="c-cMmcXp"
>
  hola
</div>
`;

exports[`Box passes \`data-\` attributes through to the DOM 1`] = `
<div
  class="c-cMmcXp"
  data-testid="some-random-id"
>
  hola
</div>
`;

exports[`Box renders as passed HTML element 1`] = `
<h1
  class="c-cMmcXp"
>
  hola
</h1>
`;

exports[`Box renders with css prop 1`] = `
<div
  class="c-cMmcXp c-cMmcXp-iiMKLbU-css"
>
  hola
</div>
`;

exports[`Box renders with theme-aware values from Stitches config 1`] = `
<div
  class="c-cMmcXp c-cMmcXp-ibcCYNQ-css"
>
  hola
</div>
`;
